src/sharding/Shard.js

```diff
@@ -190,7 +190,8 @@ class Shard extends EventEmitter {
       };
       this.process.on('message', listener);
 
-      this.send({ _eval: script }).catch(err => {
+      const _eval = typeof script === 'function' ? `(${script})(this)` : script;
```

Comment:
When the broadcastEval from a client reaches this method, the function will already be stringified to `undefined`.
(`JSON.stringify(function() {}); // undefined`)

Moving your line to `ClientShardUtil#broadcastEval` seems to work fine however.